### PR TITLE
Optimise queries in webapp

### LINKED ISSUE #940: Optimise SQL in WebApp
Issue raised by: [developer]

### What?
Optimise the sql queries in the various view functions of the webapp.

### Where?
Where is the issue likely to be (be as specific as possible)

### How?
Slow queries resulting in slow WebApp for example
image
image

### Reproducible?
[Yes/No]

### How to test the issue is resolved
A set of instructions / A script / Include test files if required

### PR DESCRIPTION
**Pull request by keiranjprice101**
### Summary of work
Combines some queries into single queries, casts a queryset to a list to prevent requerying for instrument summary and run summary pages

### How to test your work
Deploy the webapp on master branch.
View the instrument summary and run summary pages (HRPD has a lot of runs so is a good benchmark 78938 especially for run summary page)
Make note of how many queries and time in admin side panel
Redeploy this branch and visit the same pages and view the query differences.

### Additional comments
 

Fixes #940

**Before merging ensure the release notes have been updated**

### CODE REVIEW

File: WebApp/autoreduce_webapp/templates/run_summary.html [hunk 1]
```diff
@@ -12,7 +12,7 @@
         <div class="row">
             <div class="col-md-12 text-center">
                 <h2>Reduction Job #{{ run.title }}</h2>
-                {% if history.count > 1 %}
```
Comment (keiranjprice101) [inline WebApp/autoreduce_webapp/templates/run_summary.html:15]: This would previously retrigger a database query as well

**Comment by dtasev** [comment]
Huge difference, I'll let these pictures speak for themselves:

Before: 
image

After:
image

**Review by rickybassom — APPROVED** [review]
Also noticing a speed up. Tested and code looks good :+1:

**Comment by dtasev** [comment]
Looking at the SQL queries it seems we can about halve the time again if we can avoid the second SQL query to the ReductionRun table

```
SELECT ••• FROM `reduction_viewer_reductionrun` INNER JOIN `reduction_viewer_instrument` ON (`reduction_viewer_reductionrun`.`instrument_id` = `reduction_viewer_instrument`.`id`) INNER JOIN `reduction_viewer_experiment` ON (`reduction_viewer_reductionrun`.`experiment_id` = `reduction_viewer_experiment`.`id`) INNER JOIN `reduction_viewer_status` ON (`reduction_viewer_reductionrun`.`status_id` = `reduction_viewer_status`.`id`) WHERE (`reduction_viewer_instrument`.`name` = 'MARI' AND `reduction_viewer_reductionrun`.`run_number` = 27709) ORDER BY `reduction_viewer_reductionrun`.`run_version` DESC
9.36	

SELECT ••• FROM `reduction_viewer_reductionrun` WHERE (`reduction_viewer_reductionrun`.`instrument_id` = 25 AND `reduction_viewer_reductionrun`.`run_number` = 27709 AND `reduction_viewer_reductionrun`.`run_version` = 0) LIMIT 21
2.64	
```

I had a look but it's not immediately obvious where the second call happens within the `run_summary` call and if it can be avoided

**Review by dtasev — APPROVED** [review]

**Pull request merged**